tests/shell/runtime/test_shell_pty_core.py

```
@@ -262,6 +264,39 @@ def test_output_processor_filters_user_command_echo_before_command_output():
     assert rendered == b"/tmp/project\r\n"
 
 
+def test_setup_pty_reuses_manager_startup_handshake(monkeypatch):
+    class _StartedPTYManager:
+        def __init__(self, *, rows: int, cols: int, cwd: str, use_output_thread: bool):
+            assert rows == 24
+            assert cols == 80
+            assert cwd == "/tmp/frontend"
```

Ruff (0.15.10)

[error] Probable insecure usage of temporary file or directory: "/tmp/frontend"

(S108)